pkg/codegen/go/gen.go

```diff
@@ -1310,6 +1413,14 @@ func (pkg *pkgContext) getDefaultValue(dv *schema.DefaultValue, t schema.Type) (
 			return "", err
 		}
 		val = v
+		switch t.(type) {
+		case *schema.EnumType:
+			typeName := strings.TrimSuffix(strings.TrimSuffix(pkg.typeString(t), "Input"), "Ptr")
```

Comment:
It feels like there is some deficiency in typeString here. Perhaps the unwrapping of Input(T) -> T and *T -> T can happen as a transform of `t`, not the string, before we pass it to `pkg.typeString()` and pkg.typeString could take care of `pulumi.Any` as well?